Jenkinsfile

```diff
@@ -0,0 +1,155 @@
+#!groovy
+
+def isPullRequest() {
+    return binding.variables.containsKey('GITHUB_PR_NUMBER')
+}
+
+def reportResultToGithub() {
+    step([
+        $class: 'GitHubPRBuildStatusPublisher',
+        statusMsg: [content: "${GITHUB_PR_COND_REF} run ended"],
+        unstableAs: 'FAILURE'
+    ])
+}
+
+def sendMetrics(String metric, String value) {
+    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: '5b8ad2d9-61a4-43b5-b4df-b8ff6b1f16fa', passwordVariable: 'influx_pass', usernameVariable: 'influx_user']]) {
+        sh "curl -i -XPOST 'https://greatscott-pinheads-70.c.influxdb.com:8086/write?db=realm' --data-binary '${metric} value=${value}i' --user '${env.influx_user}:${env.influx_pass}'"
+    }
+}
+
+def sendABIMetric(String abi) {
+    def value = readFile("abi_$abi")
+    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: '5b8ad2d9-61a4-43b5-b4df-b8ff6b1f16fa', passwordVariable: 'influx_pass', usernameVariable: 'influx_user']]) {
+        sh "curl -i -XPOST 'https://greatscott-pinheads-70.c.influxdb.com:8086/write?db=realm' --data-binary 'abi_size,type=${abi} value=${value}i' --user '${env.influx_user}:${env.influx_pass}'"
+    }
+}
+
+def storeJunitResults(String path) {
+    step([
+        $class: 'JUnitResultArchiver',
+        testResults: path
+    ])
+}
+
+def collectAarMetrics() {
+    dir('realm/realm-library/build/outputs/aar') {
+        sh 'unzip realm-android-library-release.aar -d unzipped'
+        sh 'find $ANDROID_HOME -name dx | sort -r | head -n 1 > dx'
+        sh '$(cat dx) --dex --output=temp.dex unzipped/classes.jar'
+        sh 'cat temp.dex | head -c 92 | tail -c 4 | hexdump -e \'1/4 "%d"\' > methods'
+        sendMetrics('methods', readFile('methods'))
+
+        sh 'du -k realm-android-library-release.aar | echo -n `cut -f 1` > aar_size'
+        sendMetrics('aar_size', readFile('aar_size'))
+
+        sh '''for path in unzipped/jni/* ; do
+                abi_size=`du -k $path | cut -f 1`
+                abi=`basename $path`
+                echo -n $abi_size > abi_$abi
+              done
+        '''
+        sendABIMetric('armeabi')
+        sendABIMetric('armeabi-v7a')
+        sendABIMetric('arm64-v8a')
+        sendABIMetric('mips')
+        sendABIMetric('x86')
+        sendABIMetric('x86_64')
+    }
+}
+
+{ ->
```

Comment:
From what I understand, this actually forces the main job (coordinator) to run on the Jenkins master. If instead, you replace this closure by a `node()` call, some slave will be the coordinator. If we start using this a lot, we might overload master, so scheduling it on a node might make sense.

Maybe we should have a coordinator tag, and give it to a bunch of slaves that have good uptime/availability?